root: Add 6.22.02; remove preferred label

ChristianTackeGSI opened the pull request:
Maintainer-Ping: @chissg, @HadrienG2, @drbenmorgan, @vvolkl

HadrienG2 reviewed (approved):
LGTM

adamjstewart commented on var/spack/repos/builtin/packages/root/package.py:
```diff
@@ -28,6 +28,8 @@ class Root(CMakePackage):
     # Development version (when more recent than production).
 
     # Production version
+    version('6.22.04', sha256='a2f066d85db8eb5b5f1c72573923484a6782a47b94954eff64879609f360e951')
```
Should we update the latest production version to be preferred?

vvolkl replied:
Good question - with ROOT, it's usually wise to stay a bit behind the latest release. I asked around what the status of production ROOT is, and apparently:

> There isn't yet a universally accepted version of ROOT 6.22 . LHCb does not use it but Gaudi tests were failing until 6.22/04. ATLAS seems happy with 6.22/00, which is a bit of a mystery because they also use Gaudi. The latest patch 6.22/06 has still severe I/O issues which impact CMS. Apparently the latters have been understood and a new tag 6.22/08 is expected for the next days.

On the other hand, these issues that are usually edge cases due to the massive complexity, and won't affect most users. Users that care about production stability will most likely select a specific version of root anyway. So no strong opinion what the default should be.

ChristianTackeGSI replied:
No strong opinion from me either.

Latest 6.20 happily coincides with what we currently use.

I see two ways forward:
* Drop the `preferred` label completely and make the latest version the default. People needing specific versions will anyway pin them.
* Mark a "quite stable" version. This seems to be latest 6.20.

Both is fine with me.

vvolkl replied:
The ROOT developers would probably wany more people using the latest versions and reporting bugs. So I'd suggest to drop the `preferred` label.

vvolkl commented:
@adamjstewart Only tangentially related to this PR: Now that all hep package recipes have `tags = ['hep']`, can we apply also a label 'hep' to the PRs that change them? I guess it should even be possible to setup a workflow to do this automatically.

adamjstewart commented:
Yes, that should be possible. The spack bot I demoed at our SC20 BoF could be made to automatically add this label.

ChristianTackeGSI commented:
Marking WIP.
6.22.04 has been removed.
> 6.22/04 has a major bug, it's being fixed; 6.22/06 will be announced.

So I'll wait a few days and add 6.22.06 then.

ChristianTackeGSI commented:
So we currently have one vote for removing the label, and everyone else "has no strong opinion".

I have just pushed an intermediate state of the branch as a "WIP": Dropping 6.22.04, and dropping `preferred` (and rebased).

HadrienG2 commented:
One more vote for "don't use `preferred` if you have no strong reason for it" here ;)

ChristianTackeGSI committed:
root: Add 6.22.02, remove preferred label

Drop the preferred label from 6.20.x.
Let's just default to the latest (production) version.

ChristianTackeGSI commented:
6.22.06 seems to take a bit longer.
Let's add 6.22.02, and remove the preferred label, now.

(I rebased and reworded. No change in content.)

drbenmorgan commented:
O.k., this looks good now modulo remaining checks. Unless there are objections, I'll merge later today.
